test/fixtures/profiles/only_if/skip-control/controls/skip-control.rb

```diff
@@ -0,0 +1,56 @@
+control "control-start" do
+  desc "This control should not get skipped"
+  describe "a string" do
+    it { should cmp "a string" }
+  end
+end
+
+control "control-skip-no-message" do
+  desc "This control should get skipped"
+  only_if { false }
+  describe "a string" do
+    it { should cmp "a string" }
+  end
+end
+
+control "control-skip-with-message" do
+  desc "This control should get skipped"
+  only_if("here is a message") { false }
+  describe "a string" do
+    it { should cmp "a string" }
+  end
+end
+
+control "control-skip-test-body" do
+  desc "This control should demo that the test body does not get evaluated"
+  only_if { false }
+  describe "infinity" do
+    it { should cmp 1/0 }
+  end
+end
+
+control "control-skip-test-outer-error" do
+  desc "This control should demo that following test resources do not get evaluated"
+  only_if { false }
+  describe 1/0 do # does not error!
+    it { should cmp 1/0 }
+  end
+end
+
+control "control-skip-test-outer-error-test-first" do
+  desc "This control should demo that preceding test resources DO get evaluated"
+  describe 1/0 do # does error!
```

Comment:
At the moment there are a number of InSpec features that don't work well with Ruby statements that InSpec will need to evaluate (like `1/0` or `1+1`). I'd prefer not to put a test like this into our supported tests.

The approach I'd recommend to people is to keep *all* code inside of a resource; even if it means needing custom resources that boil down to a single line of functional code.

It's something we're chatting with @kekaichinose about, to decide how to improve on this.

Reply:
It appears the action for me the contributor is to remove the test, which is what I have done.